### DISCUSSION TITLE
Unable to GET single record from "users" collection

### DISCUSSION THREAD

**Discussion by rmso27** [pocketbase/pocketbase#5443] [Q&A] [answered]
Hi,

I'm trying to get data from a single user record and I'm not able to do so. When I call http://127.0.0.1:8090/api/collections/users/records/6ptakfsdinln9wa, I get the following result:

```
{
  "code": 404,
  "message": "The requested resource wasn't found.",
  "data": {

  }
}
```

If I try to get the list with all records with http://127.0.0.1:8090/api/collections/users/records/, I get the following result:

```
{
  "page": 1,
  "perPage": 30,
  "totalItems": 2,
  "totalPages": 1,
  "items": [
    {
      "collectionId": "_pb_users_auth_",
      "collectionName": "users",
      "created": "2024-08-30 13:27:37.934Z",
      "emailVisibility": false,
      "group": "test group",
      "id": "6ptakfsdinln9wa",
      "name": "Ruben Oliveira",
      "updated": "2024-08-30 15:21:00.303Z",
      "username": "ruben.oliveira",
      "verified": false
    },
    {
      "collectionId": "_pb_users_auth_",
      "collectionName": "users",
      "created": "2024-08-30 15:11:24.268Z",
      "emailVisibility": false,
      "group": "test group",
      "id": "9703x3urpcr8672",
      "name": "Rodrigo Oliveira",
      "updated": "2024-08-30 15:21:08.941Z",
      "username": "rodrigo.oliveira",
      "verified": false
    }
  ]
}
```

As you can see, the ID that I'm trying to read exists.

I'm running PocketBase 0.22.20.

Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
@rmso27 As mentioned in your earlier opened issue, double check your users collection View API rule.

You'll get 404 response if your client request doesn't satisfy it. Per the note in the [documentation](https://pocketbase.io/docs/api-rules-and-filters/):

> Because of the above, the API will return 200 empty items response in case a request doesn't satisfy a listRule, 400 for unsatisfied createRule and **404 for unsatisfied viewRule, updateRule and deleteRule**.
> All rules will return 403 in case they were "locked" (aka. admin only) and the request client is not an admin.

If you are sure that your client request satisfies the View API rule then provide a code sample how do you send the request.

If you are using Next.js make sure that what you are experiencing is not caused by its default cache, aka. test your request using Postman, Insomnia, curl, etc.

**Please don't treat the issues tracker and discussions as llm chat and try to provide as much details as possible to understand the bigger picture and ideally some **minimal** code sample or steps to reproduce.**

**Reply by rmso27** [reply]
Thank you very much! The rule was causing the issue!

And sorry! I'll be more careful next time before raising an issue :)
